test/collections.js

```diff
@@ -123,6 +123,11 @@ describe('Collections', function () {
       expect(map2).to.have.entries([['a', 'b'], ['c', 'd']]);
     });
 
+    it('non-entry objects should throw', function () {
+      expect(new Map([1, 2, 3])).to['throw'](TypeError);
+      expect(new Map('123')).to['throw'](TypeError);
```

Comment:
This should be the same as `new Map(['1', undefined], ['2', undefined], ['3', undefined])`